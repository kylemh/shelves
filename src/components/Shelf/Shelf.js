import React from 'react'
import PropTypes from 'prop-types';
import CollectionItem from '../CollectionItem/CollectionItem';
import CollectionItemPropTypes from '../CollectionItem/CollectionItemPropTypes';
import styles from './Shelf.module.scss';

class Shelf extends React.Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    releases: PropTypes.arrayOf(PropTypes.shape({
      ...CollectionItemPropTypes,
    })),
  };

  static defaultProps = {
    releases: [],
  };

  render() {
    const { props } = this;

    return (
      <section className={styles.Shelf}>
        <h2>Shelf</h2>
        <div className={styles.itemsContainer}>
          {props.releases.map(release => (
              <CollectionItem {...release} key={release.id} />
            )
          )}
        </div>
      </section>
    );
  }
}

export default Shelf

