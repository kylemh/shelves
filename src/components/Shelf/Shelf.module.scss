// @import '../../styles/mixins.scss';
@import '../../styles/variables.scss';

.Shelf {
  background-color: $feint;
  border-radius: 10px;
  margin: 1rem 0;
  width: 100%;
}

.container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 1rem;
}

.name {
  max-width: 50%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  height: 40px;
}

.itemsContainer {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  height: 11rem;
  padding: 1rem 0;
  margin-bottom: 0.5rem;
  overflow-x: auto;
  overflow-y: hidden;
}

.icon {
  height: $iconHeight;
}
