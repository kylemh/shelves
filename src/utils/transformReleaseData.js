/**
 * @description Pass a release to this function and get back the relevant info for this UI
 * @param {object} release
 * @param {number} release.id
 * @param {object} release.basic_information
 * @param {number} release.basic_information.year
 * @param {string} release.basic_information.title
 * @param {[{ name: string }]} release.basic_information.artists
 * @exports
 * @returns
 */
function transformReleaseData({ id, basic_information: basicInfo }) {
  const { artists, title, year } = basicInfo;

  const primaryArtistName = artists[0].name;

  return {
    id,
    primaryArtistName,
    title,
    year,
  };
}

export default transformReleaseData;
