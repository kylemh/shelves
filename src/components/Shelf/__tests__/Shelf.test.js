import React from 'react';
import { shallow } from 'enzyme';

import transformReleaseData from '../../../utils/transformReleaseData';
import mockedReleases from '../../../api/mocks/releases-0.json';
import Shelf from '../Shelf';

const testData = mockedReleases.releases.slice(0, 3).map(release => transformReleaseData(release));

describe('Shelf', () => {
  it('should render', () => {
    const wrapper = shallow(<Shelf name="Test Shelf" releases={testData} />);

    expect(wrapper).toMatchSnapshot();
  });
});
