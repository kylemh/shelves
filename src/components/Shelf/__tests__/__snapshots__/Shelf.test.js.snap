// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shelf should render 1`] = `
<section
  className="Shelf"
>
  <div
    className="container"
  >
    <EditableField
      className="name"
      initialValue="Test Shelf"
      setValueCallback={[Function]}
      tag="h3"
    />
    <Button
      disabled={false}
      fullWidth={false}
      onClick={[Function]}
      tabIndex={0}
      theme="primary"
      type="button"
    >
      <ScreenReaderOnly>
        Delete Shelf
      </ScreenReaderOnly>
      <ReactComponent
        className="icon"
      />
    </Button>
  </div>
  <DroppableReleasesList
    droppableId="some-super-duper-unique-id-123"
    releases={
      Array [
        Object {
          "id": 76035,
          "primaryArtistName": "Swayzak",
          "title": "Dirty Dancing",
          "year": 2002,
        },
        Object {
          "id": 198657,
          "primaryArtistName": "Jazkamer",
          "title": "Sound Of Music",
          "year": 2002,
        },
        Object {
          "id": 619145,
          "primaryArtistName": "Tiga",
          "title": "Sexor",
          "year": 2006,
        },
      ]
    }
  />
</section>
`;
